Gemfile.lock

```diff
@@ -1,153 +1,20 @@
 PATH
   remote: .
   specs:
-    database_cleaner (1.8.2)
```

Comment:
Can you explain the value in gitignoring Gemfile.lock in library repos? I've never been entirely clear on that. I'm guessing its that future test runs will pick up newly released versions of dependencies, giving us a heads-up if one of them fails?

Reply:
Personally, I've always weighed this against build/failure reproducibility across development instances and found the latter more compelling, but I'd love to learn something new here. What are your views?

Reply:
It prevents builds from staying in sync with reality. Your gain stability in your builds by locking to a period of time rather than versions. If you have specific versions that you need for a build to pass they should be in your `gemspec`, or if you need some custom magic in your `Gemfile`.

Someone who consumes the library will not get the versions specified in your `Gemfile` or `Gemfile.lock` no matter how they get the library, so you end up not seeing failures that affect them until someone bothers to open an issue.

Reply:
@JonRowe Ah, that makes sense, and its a particularly relevant argument, considering the amount of effort this library is currently expending to address exactly that situation, i.e. tied to a 2010 reality!

I wonder though, does the existence of dependabot completely change this situation? Now that its on the job, do we get the best of both worlds by leaving Gemfile.lock checked in? Developers of the library don't suffer from works-on-my-machine, and dependabot ensures that we stay in sync with reality?

Reply:
Dependabot isn't up to that yet. It too often misses things because theres no security implication, or ignores things it can't update.

Reply:
Ah, well, that's a bummer. Maybe some day. I'll remove the .lock files from git before the release.